app/src/main/kotlin/com/wire/android/ui/home/conversations/DownloadedAssetDialog.kt

```diff
@@ -0,0 +1,62 @@
+package com.wire.android.ui.home.conversations
+
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import com.wire.android.R
+import com.wire.android.ui.common.WireDialog
+import com.wire.android.ui.common.WireDialogButtonProperties
+import com.wire.android.ui.common.WireDialogButtonType
+import com.wire.android.util.permission.WriteToExternalStorageRequestFlow
+
+@Composable
+fun DownloadedAssetDialog(
+    conversationViewState: ConversationViewState,
+    onSaveFileToExternalStorage: (String?, ByteArray) -> Unit,
+    onOpenFileWithExternalApp: (String?, ByteArray) -> Unit,
+    hideOnAssetDownloadedDialog: () -> Unit
```

Comment:
minor thing: but now we are kind of controlling the UI both from the ViewModel by calling directly the function inside the VIewModel closing the dialog and also from within the Dialog, I think it would be better to call this function like. : 
onDownloadCancelled()